### PR TITLE
c8d: Use the labels to get the children of an image

### PR DESCRIPTION
**Pull request by rumpl**
**- What I did**

Extracted a part of https://github.com/moby/moby/pull/46634

Instead of looking at the rootfs of the images, we only look at the parent label. `Children` also now return only the direct children, which puts it in line with what graph drivers do

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by laurazard — APPROVED** [review]
LGTM

File: daemon/containerd/image_children.go [hunk 1]
```diff
@@ -15,67 +15,16 @@ import (
...
-		parentRootFS = append(parentRootFS, rootfs)
-	}
-
-	imgs, err := i.client.ImageService().List(ctx)
+	imgs, err := i.client.ImageService().List(ctx, "labels."+imageLabelClassicBuilderParent+"=="+string(id))
 	if err != nil {
 		return []image.ID{}, errdefs.System(errors.Wrap(err, "failed to list all images"))
 	}
 
 	children := []image.ID{}
```
Comment (laurazard) [inline daemon/containerd/image_children.go:25]:
```suggestion
	var children []image.ID
```

Comment (rumpl) [reply daemon/containerd/image_children.go]: f-f-f-fixed

**Review by vvoland — APPROVED** [review]

File: daemon/containerd/image_children.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/containerd/image_children.go]: I guess another linter would say `children := make([]image.ID, 0, len(imgs))` 😉

**Pull request merged**